src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -19,91 +21,129 @@ class PaymentService {
     Znajdź i zwróć płatności posortowane po dacie rosnąco
      */
     List<Payment> findPaymentsSortedByDateAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .sorted(Comparator.comparing(Payment::getPaymentDate))
+                .collect(Collectors.toList());
     }
 
+
     /*
     Znajdź i zwróć płatności posortowane po dacie malejąco
      */
     List<Payment> findPaymentsSortedByDateDesc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .sorted(Comparator.comparing(Payment::getPaymentDate).reversed())
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów rosnąco
      */
     List<Payment> findPaymentsSortedByItemCountAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .sorted(Comparator.comparingInt(i -> i.getPaymentItems().size()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów malejąco
      */
     List<Payment> findPaymentsSortedByItemCountDesc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .sorted((s1, s2) -> Integer.compare(s2.getPaymentItems().size(), s1.getPaymentItems().size()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności dla wskazanego miesiąca
      */
     List<Payment> findPaymentsForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .filter(p -> p.getPaymentDate().getMonth() == yearMonth.getMonth())
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności dla aktualnego miesiąca
      */
     List<Payment> findPaymentsForCurrentMonth() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .filter(p -> p.getPaymentDate().getMonth() == dateTimeProvider.zonedDateTimeNow().getMonth())
+                .collect(Collectors.toList());
     }
```

Comment:
Zamiast pisać jeszcze raz, to możesz wywołać `findPaymentsForGivenMonth` z odpowiednim parametrem